spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -102,19 +102,21 @@ def get_parser():
             "template label. No scaling will be performed. \n"
             "\n"
             "If two labels are provided, a linear transformation (translation + rotation + superior-inferior linear "
-            "scaling) will be applied. The strategy here is to defined labels that cover the region of interest. For "
+            "scaling) will be applied. The strategy here is to define labels that cover the region of interest. For "
             "example, if you are interested in studying C2 to C6 levels, then provide one label at C2 and another at "
             "C6. However, note that if the two labels are very far apart (e.g. C2 and T12), there might be a "
             "mis-alignment of discs because a subject''s intervertebral discs distance might differ from that of the "
             "template.\n"
             "\n"
-            "If more than two labels (only with the parameter '-disc') are used, a non-linear registration will be "
-            "applied to align the each intervertebral disc between the subject and the template, as described in "
-            "sct_straighten_spinalcord. This the most accurate and preferred method. This feature does not work with "
-            "the parameter '-ref subject', where only a rigid registration is performed.\n"
+            "If more than two labels are used, a non-linear registration will be applied to align the each "
+            "intervertebral disc between the subject and the template, as described in "
+            "sct_straighten_spinalcord. This the most accurate and preferred method. Note: \n"
+            "  - Using more than two labels will only work with disc labels (i.e. '-ldisc')\n"
```

Comment:
is that true? i'm wondering why it wouldn't work with -l labels 🤔

Reply:
Good question -- I was just going off of what was written previously ("`If more than two labels (only with the parameter '-disc') are used`") but I can double-check!

Reply:
I followed the tutorial steps, swapping out the typical two-label `sct_label_utils` step (`3,9`) with this:

```
sct_label_utils -i t2_seg_labeled.nii.gz -vert-body 3,4,5,6,7,8,9 -o t2_labels_vert.nii.gz
```

When running `sct_register_to_template`, I get this error:

```
ERROR: Label file has 7 label(s). It must contain one or two labels.
```

The error message comes from this part of the codebase:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/8985ba391ff149cb7af8a086dd7d0d28be90f010/spinalcordtoolbox/scripts/sct_register_to_template.py#L916-L918

However, since it's just a `printv` statement, the error doesn't actually halt the program... so it keeps going! (Log [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/files/7591237/sct_register_to_template_with_7_labels.txt).)

Results:

template2anat.nii.gz (overlaid on t2.nii.gz):

![template2anat](https://user-images.githubusercontent.com/16181459/143135914-8e35cc32-9184-4133-bd34-2f2c8db70a8a.png)

anat2template.nii.gz (overlaid on PAM50_t2.nii.gz):

![anat2template](https://user-images.githubusercontent.com/16181459/143136356-cffae9a0-2053-46dc-b99b-24c23d061227.png)

Apart from the seemingly spurious error message (specifically for `body` labels), it does seem like it works fine? I wonder why that error message was put there in the first place.

Reply:
The context here is from a commit made 3 years ago: https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/e64928c4f976f18351a67b95333867984533a25e

It looks like we used to only permit exactly 2 labels, then we opened it up to 1 or 2. But we've never seemed to permit 2+ labels... :thinking:

Reply:
thank you for these investigations, @joshuacwnewton . I _think_ that this error message was set because the motivation for allowing more than 2 labels was based on older investigations when using multiple disc labels-- given that it was motivated by disc labeling, we did not really consider if this could also apply to vert labels (and since we didn't want to deal with too many things at that time we just decided to say "let's authorize this only for disc labels"). but is looks like we can now remove that limitation